libpod/oci_linux.go

```diff
@@ -28,7 +28,7 @@ func (r *OCIRuntime) moveConmonToCgroup(ctr *Container, cgroupParent string, cmd
 				logrus.Warnf("Failed to add conmon to systemd sandbox cgroup: %v", err)
 			}
 		} else {
-			cgroupPath := filepath.Join(ctr.config.CgroupParent, fmt.Sprintf("libpod-%s", ctr.ID()), "conmon")
+			cgroupPath := filepath.Join(ctr.config.CgroupParent, "conmon")
```

Comment:
Where does this get cleaned up now?

Reply:
If we're in a pod, when the pod's CGroup is removed as the pod itself is removed. If we're not in a pod, never - but leaking one total CGroup per system doesn't seem terrible.

We could do instance-counting for containers sharing a CGroup parent and clean it up after all of them exit, but that seems messy and potentially racy.

Reply:
> ... but leaking one total CGroup per system doesn't seem terrible

It's per-configured-`CgroupParent`.  I'm not sure how often that differs from "per system", although it's probably not often.

> We could do instance-counting for containers sharing a CGroup parent and clean it up after all of them exit, but that seems messy and potentially racy.

The kernel already counts instances for us by tracking PIDs in the cgroup.  I was just thinking we'd try to delete this in `libpod delete ...` and log warn-level messages on failure.

> ... but that seems messy and potentially racy.

Perhaps the race would be deleting one container removing the unused crgoup after a parallel creation ensured the directory existed but had not yet moved the conmon process into it.  That seems like a narrow race, but something that was still worth protecting with a lock.  I'm also fine punting this sort of thing to follow-up work, because leaking this cgroup seems like a lesser evil than our current leaks due to cgroup nesting ;)